x/assets/types/genesis.go

```diff
@@ -25,13 +34,344 @@ func NewGenesis(
 // for any unit / integration tests.
 func DefaultGenesis() *GenesisState {
 	return NewGenesis(
-		DefaultParams(), []ClientChainInfo{}, []StakingAssetInfo{}, []DepositsByStaker{},
+		DefaultParams(), []ClientChainInfo{}, []StakingAssetInfo{}, []DepositsByStaker{}, []AssetsByOperator{},
 	)
 }
 
+// CommonValidation is used to check for duplicates in the input list
+// and validate the input information simultaneously.
+// It might be used for validating most genesis states.
+func CommonValidation[T any, V constraints.Ordered, D any](
```

Comment:
Love it! Maybe move it to `utils/utils.go` because it can be reused across modules?

Reply:
Thanks, I will move it to `utils/utils.go` in the next commit.